filebeat/harvester/reader/message.go

```diff
@@ -0,0 +1,22 @@
+package reader
+
+import (
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+// Message represents a reader event with timestamp, content and actual number
+// of bytes read from input before decoding.
+type Message struct {
+	Ts      time.Time     // timestamp the content was read
+	Content []byte        // actual content read
+	Bytes   int           // total number of bytes read to generate the message
+	Fields  common.MapStr // optional fields that can be added by reader
+}
+
+func NewMessage() Message {
+	return Message{
+		Fields: common.MapStr{},
```

Comment:
That seems to be the better solution. We could even think about making Fields private.